src/rgw/rgw_op.cc

```diff
@@ -5110,6 +5110,11 @@ void RGWCopyObj::execute(optional_yield y)
   if (init_common() < 0)
     return;
 
+  /* XXX fixup for rgw_file call path */
+  if (! s->object->get_bucket()) {
+    s->bucket = src_object->get_bucket()->clone();
```

Comment:
Maybe also s->object->set_bucket(s->bucket->get()); ?

Reply:
is that more correct?  will change

Reply:
(i mean, it seems obviously better, didn't realize it would work)

Reply:
ack, will add